addons/l10n_ec/models/account_move.py

```diff
@@ -134,44 +133,30 @@ class AccountMove(models.Model):
         string="Payment Method (SRI)",
     )
 
-    def _get_l10n_ec_identification_type(self):
+    def _get_l10n_ec_ats_identification_type(self):
+        # Helps filter out document types based on subset of Table 2 of SRI's ATS specification
         self.ensure_one()
-        move = self
-        it_ruc = self.env.ref("l10n_ec.ec_ruc", False)
-        it_dni = self.env.ref("l10n_ec.ec_dni", False)
-        it_passport = self.env.ref("l10n_ec.ec_passport", False)
-        is_final_consumer = verify_final_consumer(move.partner_id.commercial_partner_id.vat)
-        is_ruc = move.partner_id.commercial_partner_id.l10n_latam_identification_type_id.id == it_ruc.id
-        is_dni = move.partner_id.commercial_partner_id.l10n_latam_identification_type_id.id == it_dni.id
-        is_passport = move.partner_id.commercial_partner_id.l10n_latam_identification_type_id.id == it_passport.id
-        l10n_ec_is_exportation = move.partner_id.commercial_partner_id.country_id.code != 'EC'
+        idtype = self.partner_id._l10n_ec_get_identification_type()
         identification_code = False
-        if move.move_type in ("in_invoice", "in_refund"):
-            if is_ruc:
+        if self.move_type in ("in_invoice", "in_refund"):
+            if idtype == 'ruc': # includes final consumer
```

Comment:
@xmo-odoo I've moved what I could to an enum and its classmethods
Because ATS and SRI codes are not exactly the same, the code is not as straightforward as I'd like nor can I use the enum everywhere I'd like.
Because enums have limited subclassing, the EDI logic stays out of the enum (alternatives considered: using mixins + dynamic class definition; having EDI code in community)